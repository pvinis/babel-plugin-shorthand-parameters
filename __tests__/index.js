const babel = require('@babel/core')

const plugin = require('../src')


const inputs = [
`
const b = $.id
`,
`
[1,2].map($.id)
`,
`
const b = $0.id
`,
`
const b = $0.id + $1.ok
`,
`
const b = $1.ok
`,
]

it('works', () => {
	inputs.map(input => {
		const { code } = babel.transform(input, {plugins: [plugin]})
		expect(code).toMatchSnapshot()
	})
})
