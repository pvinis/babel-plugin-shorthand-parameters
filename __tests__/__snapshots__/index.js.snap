// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `"const b = x => x.id;"`;

exports[`works 2`] = `"[1, 2].map(x => x.id);"`;

exports[`works 3`] = `"const b = x0 => x0.id;"`;

exports[`works 4`] = `"const b = (x0, x1) => x0.id"`;

exports[`works 5`] = `"const b = (x0, x1) => x0.id + x1.ok"`;

exports[`works 6`] = `"const b = (x0, x1) => x1.ok;"`;
